part_1/src/com/chernova/library/Main.java

```diff
@@ -0,0 +1,186 @@
+                        option = -1;
+                        System.out.print("Список книг:");
+                        System.out.println(Library.getAllBooks().toString().replaceAll("^\\[|,|\\]$", ""));
+                        break;
+                    }
+
+
+                case 2:
+                    Book newBook = new Book();
+                    System.out.println("Заполните все поля для добавления книги");
+                    System.out.print("id  книги: ");
+
+                    int idAdd = -1;
+                    boolean readyToSet = false;
+                    while (!readyToSet) {
+                        try {
+                            readyToSet = true;
+                            idAdd = sc.nextInt();
+                            if (idAdd < 1) {
+                                readyToSet = false;
+                                System.out.println("id  книги должен быть положительным");
+                            }
+
+                        } catch (InputMismatchException e) {
+                            System.out.println("id это число. Введите id еще раз");
+                            readyToSet = false;
+                            sc.next();
+                        }
+                    }
+                    newBook.setId(idAdd);
+                    System.out.print("название  книги: ");
+                    sc.nextLine();
+                    newBook.setTitle(sc.nextLine());
+
+                    System.out.println("жанр  книги:");
+                    System.out.print(ANSI_YELLOW);
+                    Genre.printGenre();
+                    System.out.print(ANSI_RESET);
+
+                    int genreAdd = -1;
+                    readyToSet = false;
+                    while (!readyToSet) {
+                        try {
+                            readyToSet = true;
+                            genreAdd = sc.nextInt();
+
+                            if (genreAdd < 1 || genreAdd > Genre.values().length) {
+                                readyToSet = false;
+                                System.out.println("Выберите жанр из списка");
+                            }
+
+                        } catch (InputMismatchException e) {
+                            System.out.println("Введите номер жанра");
+                            readyToSet = false;
+                            sc.next();
+                        }
+                    }
+                    newBook.setGenre(Genre.getGenreByID(genreAdd));
+                    Library.addBook(newBook);
+                    break;
+
+                case 3:
+                    System.out.print("id  книги: ");
+                    try {
+                        int idEdit = sc.nextInt();
+                        boolean existBook = false;
+                        for (int i = 0; i < Library.list.size(); i++) {
+                            if (idEdit == Library.list.get(i).getId()) {
+                                existBook = true;
```

Comment:
jump out from cycle after editing (now, if you have 1000000 books and you want to edit first, you still need to iterate over all of them even if you already edited required)